Bugfix/2277 fix solutions management

alexandre-allard opened the pull request:
**Component**: scripts, solutions

**Context**: 

**Summary**:

**Acceptance criteria**: 

---

See: #2277

alexandre-allard committed:
scripts: activate latest version of a solution by default

If no version is provided during solution activation,
we use the latest available.

Refs: #2277

alexandre-allard committed:
scripts: ensure we use the fullpath for solution archives

We resolv the full path of archives, otherwise we
can end up with relative paths in solutions configuration
which is wrong.

alexandre-allard committed:
scripts: ensure the provided archive is an ISO

check the file type of the provided archives to
avoid adding non-ISO file to the solutions configuration.

bert-e commented:
# Hello alexandre-allard-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/2277-fix-solutions-management` into
`development/2.4`
* [w/2.5/bugfix/2277-fix-solutions-management](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/bugfix/2277-fix-solutions-management) will be merged into `development/2.5` 
* [w/2.6/bugfix/2277-fix-solutions-management](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/bugfix/2277-fix-solutions-management) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

alexandre-allard commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **approve**

alexandre-allard committed:
scripts: fix (de)activate helper message

Message was false as it was saying we deploy
UI, Operator & CRDs, but we only deploy CRDs.
Also remove --version flag from deactivate command
as we can't have multiple version activated at the
same time

TeddyAndrieux reviewed (approved):
LGTM

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/7c85acb62a6029d492e4f1f4393a82c71e9cffe2) did not succeed in branch w/2.6/bugfix/2277-fix-solutions-management.




*The following options are set:* **approve**

Ebaneck commented on scripts/solutions.sh:
```diff
@@ -261,6 +259,11 @@ configure_archives() {
     local removed=${1:-False}
 
     for archive in "${ARCHIVES[@]}"; do
+        if file "$archive" | grep -vq 'ISO 9660'; then
```
Minor and can be omitted
We are searching for the absence of a  fixed string here `ISO 9660` so we can make this faster with `fgrep`?? no??

slaperche-scality replied:
Or `grep -F` (`fgrep` is deprecated, as explained in the man page).

But I don't think we will observe a perf improvement here ^^

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



Please check the status of the associated issue None.

Goodbye alexandre-allard-scality.
